components/tabs/src/storage.rs

```diff
@@ -188,27 +211,80 @@ impl TabsStorage {
                         );
                         None
                     }
+                };
+                let mut crts: Vec<ClientRemoteTabs> = Vec::new();
+                for (record, last_modified) in records.unwrap_or_default() {
+                    let id = record.id.clone();
```

Comment:
Let's add a comment somewhere here explaining that the `id` from the database is the id from the clients collection, but we need to obtain the fxa_device_id for that client to return to the consumers, and include a link to #5199 in that `else` block (becaise we want that else block to die entirely at some point as it will be doing the wrong thing)